src/helpers/highlight.ts

```diff
@@ -2,15 +2,24 @@ import { getPropertyByPath } from '../lib/utils';
 import { TAG_REPLACEMENT } from '../lib/escape-highlight';
 import { component } from '../lib/suit';
 
+export type HighlightOptions = {
+  attribute: string;
+  highlightedTagName?: string;
+  hit: object;
```

Comment:
```suggestion
  hit: {
    _highlightResult: {
      [attribute: string]: {
        value: string;
      },
    },
  };
```

This way you don't need to do "as string" I think

Reply:
Actually I created the type `Hit` so I can reuse it here 👍

Reply:
I don't necessarily need the "as string" but `getPropertyBypath` doesn't ensure the return type. It gives any type given by the path that resolved in the object. That's why I explicitly casted it.